### ISSUE TITLE
infallible_destructuring_match misses ref in suggested fix

### ISSUE THREAD

**Issue by chevdor** [rust-lang/rust-clippy#7499]
I am using `clippy 0.1.55 (88539999 2021-07-06)`.

The suggestion given for:
```
let t1 = match type_section.types()[1] {
   Type::Function(ref func_type) => func_type
};
```
is:
```
let Type::Function(t1) = type_section.types()[1];
```

it should actually include the `ref`:
```
let Type::Function(ref t1) = type_section.types()[1];
```

**Comment by giraffate** [comment]
Can you provide a minimal reproduction on the playground?

**Comment by chevdor** [comment]
This issue appears to be fixed in `clippy 0.1.56 (49272389 2021-07-29)`

**Comment by pheki** [comment]
Also bumped into this yesterday, seems that this isn't fixed yet: [playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=%23%5Bderive(Debug)%5D%0Astruct%20Struct%3B%0A%0A%23%5Bderive(Debug)%5D%0Aenum%20Enum%20%7B%0A%20%20%20%20Variant(Struct)%2C%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20v%20%3D%20Enum%3A%3AVariant(Struct)%3B%0A%20%20%20%20let%20s%20%3D%20match%20v%20%7B%0A%20%20%20%20%20%20%20Enum%3A%3AVariant(ref%20s)%20%3D%3E%20s%0A%20%20%20%20%7D%3B%0A%20%20%20%20%0A%20%20%20%20println!(%22%7Bv%3A%3F%7D%20%7Bs%3A%3F%7D%22)%3B%0A%7D)
